scripts/DllManipulatorScript.cs

```diff
@@ -85,7 +91,7 @@ private void OnEnable()
             InitializationTime = timer.Elapsed;
         }
 
-        private void OnDestroy()
+        private void OnDisable()
```

Comment:
Now I think about it, there is no reason to have `OnDisable`, so we can leave this at `OnDestroy`. The user can manually unload the dlls when required.

Reply:
But is it always called when the assembly gets reloaded and when you start in-editor playback? Otherwise it could leak a DLL reference if user didn't perform explicit unload.

Reply:
You're right, OnDestroy is not called when exit mode is left if we use `[ExecuteInEditMode]`. I don't quite understand what you mean in your last comment. 

As a better alternative to `OnDestroy` to call Unload after everything, we can use `EditorApplication.playModeStateChanged` with the `EnteredEditMode` state: (example from the docs)

```c#
using UnityEditor;

// ensure class initializer is called whenever scripts recompile
[InitializeOnLoadAttribute] //<- not relevant for this point
public static class PlayModeStateChangedExample
{
    // register an event handler when the class is initialized
    static PlayModeStateChangedExample()
    {
        EditorApplication.playModeStateChanged += LogPlayModeState;
    }

    private static void LogPlayModeState(PlayModeStateChange state)
    {
        if(state == PlayModeStateChange.EnteredEditMode)
            // UnloadAll
    }
}
```
This is executed after all `OnDestroy`s but is in a static context. However, this should work as we have the `_singletonInstance` variable.

Reply:
In a build you would still need to use the `OnDestroy` of course, so more `#if`s.

Reply:
This seems more complicated than I thought. When the play mode is entered/left the scene is reloaded, so the instance is reset and the static variables as well. So we can't use what I stated above. 

So the ordering goes like this
```c#
//Recompile/start Unity
OnEnable

//Press Play
ExitingEditMode
OnDisable()
OnDestroy()
//Scene Reloaded and play mode entered
OnEnable()
EnteredPlayMode

//Stop Play mode
ExitingPlayMode
OnDisable()
OnDestroy()
//Scene Reloaded edit mode entered
OnEnable()
EnteredEditMode
```
This means that in fact leaving it mostly as is without `playModeStateChanged` and using `OnEnable` and `OnDestroy` is the best. The dll's will be loaded/unloaded often, but as we said below you can disable it in edit mode easily if you want.